packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -2219,6 +2221,12 @@ class TimePickerDialog extends StatefulWidget {
   /// Callback called when the selected entry mode is changed.
   final EntryModeChangeCallback? onEntryModeChanged;
 
+  /// {@macro flutter.material.time_picker.switchToInputEntryModeIcon}
+  final Icon? switchToInputEntryModeIcon;
```

Comment:
Since the default is a const icon, can we instead make these non-null and provide them as default arguments in the constructor?

I like that we provide an `Icon` widget; users can then tweak the icon, size, and color, too.

Reply:
Hi, thanks for your time. The approach you're suggesting makes much more sense. I initially coded it that way because it was done similarly in `date_picker.dart`
https://github.com/flutter/flutter/blob/0500af2b5f750925e355266360307ff4e69c0954/packages/flutter/lib/src/material/date_picker.dart#L441-L442

Reply:
+1 to having it be non-nullable, cause why not?

Thanks for changing it here!

Reply:
I would suggest nullable props so we can define defaults once instead of 3 times at three difference places.

Reply:
That's a good point—a single source of truth is easier to maintain, plus being nullable would make it parallel with date_picker.dart 👍